keyboards/ymdk/oni_hhkb/oni_hhkb.c

```diff
@@ -0,0 +1,50 @@
+/* Copyright 2025 YMDK
+ * Copyright 2025 chrisgve <https://github.com/chrisgve>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "quantum.h"
+
+#ifdef RGB_MATRIX_ENABLE
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case RM_TOGG:
```

Comment:
Code like this was removed from prior boards (e.g. Idobao's) because it does not align with base code usage and at the time was moved to vendor keymaps (later deleted/moved to vendor userspaces). The spirit of QMK code is to simply support the hardware in its simplest form. This code asserts that this is how a user wants to use this feature (since the call to the `_user` function is bypassed for `RM_TOGG` being called after the fact). It also does not account for `RM_ON` and `RM_OFF` keycodes (not saying to add them, saying it's not complete and not needed at all).